### ISSUE TITLE
Add `rb_proc_new` for creating Ruby procs in Rust

### ISSUE THREAD

**Issue by danielpclark** [danielpclark/rutie#64]
The Ruby C method `rb_proc_new` was created to [“make Proc from C function”](https://github.com/ruby/ruby/commit/b664645ddc5dc0d1dceb0b0f4a9710f0ce5a0c29).  This means we can also make a `Proc` from a Rust function.

The method signature is as follows in C:

```c
VALUE
rb_proc_new(VALUE (*func)(ANYARGS) /* VALUE yieldarg[, VALUE procarg] */, VALUE val)
{ /* ... */ }
```

Which means  the Rust representation may be one of the following:

```rust
pub fn rb_proc_new(func: CallbackPtr, args: Value) -> Value;
```

```rust
pub fn rb_proc_new(func: extern fn(Value, Option<Value>) -> Value, args: Value) -> Value;
```

I would like someone who has a more intimate knowledge of using FFI callback pointers in Rust to help create this and allow us to use Rust closures as Ruby procs.

**Comment by dsander** [comment]
Yes I used lldb as it's, as far as I know, the only debugger available on MacOS. You can do both, attach it to a rust binary or a ruby process. To have debug information available you need at least [`debug = true`](https://github.com/danielpclark/rutie/blob/master/examples/rutie_ruby_gvl_example/Cargo.toml#L13-L17) enabled for the release build.

If your Rust/Ruby process segfaults you only need to run it with lldb:

Rust:
```
$ rust-lldb <path to the binary>
....
(lldb) run
```

Ruby
```
$ lldb
(lldb) target create ~/.rbenv/versions/2.5.3/bin/ruby
(lldb) run -Ilib your-ruby-script.rb
```

You can navigate through the stack using the `up` and `down` commands.

When I want to set breakpoints I am using the `CodeLLDB` extension for VSCode with this launch config:

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "lldb",
            "request": "launch",
            "name": "Debug",
            "program": "~/.rbenv/versions/2.5.3/bin/ruby",
            "args": ["-Ilib", "run.rb"],
            "cwd": "/working/directory"
        }
    ]
}
```
